modules/reactor/src/meshgenerators/AssemblyMeshGenerator.C

```diff
@@ -0,0 +1,503 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "AssemblyMeshGenerator.h"
+
+#include "MooseApp.h"
+#include "Factory.h"
+#include "libmesh/elem.h"
+#include "MooseMeshUtils.h"
+
+registerMooseObject("ReactorApp", AssemblyMeshGenerator);
+
+InputParameters
+AssemblyMeshGenerator::validParams()
+{
+  auto params = MeshGenerator::validParams();
+
+  params.addRequiredParam<std::vector<MeshGeneratorName>>(
+      "inputs", "The PinMeshGenerators that form the components of the assembly.");
+
+  params.addRequiredParam<subdomain_id_type>("assembly_type",
+                                             "The integer ID for this assembly type definition");
```

Comment:
Not really clear
"The assembly_id integer ID for this assembly" might be more clear?

Reply:
The assembly_type is different from assembly_id. Both are extra element integers. Assembly_type is range (0, # of unique assemblies-1). Assembly_id is usually range (0, NumTotalAssemblies-1). So if there are 6 fuel assemblies and 1 control assemblies, there are only 2 assembly types (0, 1) but 7 assembly_id values. Is this the confusion? @shikhar413  @ScottRichards-ANL